src/engraving/libmscore/edit.cpp

```diff
@@ -4264,6 +4265,22 @@ void Score::restoreInitialKeySig()
         undoAddElement(newKeySig);
 
         staff->setKey(Fraction(0, 1), keySigEvent);
+
+        Segment* timeSegment = firstMeas->undoGetSegment(SegmentType::TimeSig, startTick);
+        TimeSig* newTimeSig = Factory::createTimeSig(timeSegment);
+        newTimeSig->setTrack(staff->idx() * VOICES);
+        newTimeSig->setSig(Fraction(4, 4));
+        timeSegment->add(newTimeSig);
+        undoAddElement(newTimeSig);
+
+        SigEvent timeSigEvent;
+        timeSigEvent.setTimesig(Fraction(4, 4));
+        timeSigEvent.setNominal(Fraction(4, 4));
+
+        firstMeas->setTimesig(newTimeSig->sig());
+        firstMeas->setTicks(newTimeSig->sig());
+        staff->addTimeSig(newTimeSig);
+        sigmap()->add(0, timeSigEvent);
```

Comment:
`firstMeas->setTimesig`, `firstMeas->setTicks` and `sigmap()->add` only need to be called once, instead of for every staff again. So it would be good to take them out of the for-loop. Also, I'm not sure if calling `staff->addTimeSig` and `sigmap()->add` is really necessary; it looks like that will be done in `Score::rebuildTempoAndTimeSigMaps`, which will eventually be called anyway after `Score::insertMeasure`.